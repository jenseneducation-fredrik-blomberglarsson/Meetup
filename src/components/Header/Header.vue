<template>
  <div class="header">
    <HamburgerIcon />
  </div>
</template>

<script>
import HamburgerIcon from "@/components/HamburgerIcon/HamburgerIcon.vue";
export default {
  name: "Header",
  components: {
    HamburgerIcon,
  },
};
</script>

<style>
</style>