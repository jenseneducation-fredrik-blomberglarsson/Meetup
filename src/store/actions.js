import { getMeetups } from "../data/meetupData";

const actions = {
  async fetchMeetups(context) {
    const data = await getMeetups();
    context.commit("setMeetups", data);
  },
  addThisMeetup(context, meetup) {
    context.commit('addMeetup', meetup);
  },
  removeThisMeetup(context, id) {
    context.commit('removeMeetup', id)
  },
  switchThisMeetupToggle(context, boolean) {
    context.commit('switchMeetupToggle', boolean);
  }



  /*   isBtnVisible(context) {
      context.commit('isVisible');
    } */
}

export default actions;