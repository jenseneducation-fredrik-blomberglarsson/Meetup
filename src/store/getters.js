const getters = {
  getMeetups: state => {
    return state.meetups;
  },
  getJoinedMeetups: state => {
    return state.joinedMeetups;
  },
  getMeetupToggle: state => {
    return state.meetupToggle;
  }
}


export default getters;