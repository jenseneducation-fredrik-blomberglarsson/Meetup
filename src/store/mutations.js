const mutations = {
  setMeetups(state, meetups) {
    state.meetups = meetups;
  }
}

export default mutations;