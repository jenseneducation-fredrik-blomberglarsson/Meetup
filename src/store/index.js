import Vue from 'vue'
import Vuex from 'vuex'
import actions from './actions.js';
import mutations from './mutations.js';
import getters from './getters.js';

Vue.use(Vuex)

//Global state
export default new Vuex.Store({
  state: {
    meetups: [],
    joinedMeetups: [],
    previousMeetups: [],
    meetupToggle: false,
  },

  actions: actions,
  mutations: mutations,
  getters: getters
});