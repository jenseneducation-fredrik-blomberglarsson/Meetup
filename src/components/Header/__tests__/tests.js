import { shallowMount, mount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import store from '../../../store/index.js';
import Header from '@/components/Header/Header.vue';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe('The page is rendered', () => {
  test('Test so that everything renders properly', async () => {
    const wrapper = shallowMount(Header, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  })
  test('testing if router-link for front-page exists in component Header', () => {
    const wrapper = mount(Header, {
      store,
      localVue,
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.findComponent(RouterLinkStub).props().to).toBe('/');
  });
})