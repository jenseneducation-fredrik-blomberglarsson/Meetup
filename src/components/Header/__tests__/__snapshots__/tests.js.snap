// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page is rendered Test so that everything renders properly 1`] = `
<div
  class="header"
>
  <div
    id="nav"
  >
    <router-link-stub
      ariacurrentvalue="page"
      class="router-link"
      event="click"
      tag="a"
      to="/"
    >
      Home
    </router-link-stub>
    |
    
    <router-link-stub
      ariacurrentvalue="page"
      class="router-link"
      event="click"
      tag="a"
      to="/meetups"
    >
      Upcoming Meetups
    </router-link-stub>
     
    <router-link-stub
      ariacurrentvalue="page"
      class="router-link"
      event="click"
      tag="a"
      to="/mymeetups"
    >
      My Meetups
    </router-link-stub>
  </div>
</div>
`;
